rag-vector-sveltekit-xata.mdx

```diff
@@ -0,0 +1,397 @@
+}
+```
+
+Using Vercel’s AI SDK baked-in `experimental_buildLlama2Prompt` function, we’re able to:
+
+1.  Pass instructions so that the code blocks in the markdown-like responses generated by AI are prepended with \`\`\`\`\`. This'll help us highlight them in our frontend setup via a syntax highlighter. This is done via creating a message with the role of `assistant`and passing our relevant instructions in the`content` field.
+2.  Pass the dynamically generated context (as string) for the AI to consider before generating responses. This is done via creating a message with the role of `system` and passing our relevant context string in the `content` field.
+3.  Pass the whole conversation between user and chatbot as `messages`, which contains each message with it’s relevant `content` and `role` field.
+
+````typescript
+import { experimental_buildLlama2Prompt } from 'ai/prompts';
+
+// Now use Replicate LLAMA 70B streaming to perform the autocompletion with context
+const response = await replicate.predictions.create({
+  // You must enable streaming.
+  stream: true,
+  // The model must support streaming. See https://replicate.com/docs/streaming
+  model: 'meta/llama-2-70b-chat',
+  // Format the message list into the format expected by Llama 2
+  // @see https://github.com/vercel/ai/blob/99cf16edf0a09405d15d3867f997c96a8da869c6/packages/core/prompts/huggingface.ts#L53C1-L78C2
+  input: {
+    prompt: experimental_buildLlama2Prompt([
+      {
+        // create a system content message to be added as
+        // the llama2prompt generator will supply it as the context with the API
+        role: 'system',
+        content: systemContext
+      },
+      {
+        // create a system instruction
+        // make sure to wrap code blocks with ``` so that the svelte markdown picks it up correctly
+        role: 'assistant',
+        content: `When creating repsonses sure to wrap any code blocks that you output as code blocks and not text so that they can be rendered beautifully.`
+      },
+      // also, pass the whole conversation!
+      ...messages
+    ])
+  }
+});
+````
+
+## Wiring Up Chatbot UI in SvelteKit
+
+Using Vercel's AI SDK we’re able to save tremendous time in setting up a reactive chatbot UI in SvelteKit. The `useChat` utility from `ai/svelte` encapsulates the frontend logic to maintaining the array of messages exchanged between the user and the chatbot, submitting form responses to the relevant API endpoint with the messages exchanged, and stream content right into the UI from the streaming API response.
+
+```html
+<script>
+  // File: src/routes/+page.svelte
+  import { useChat } from 'ai/svelte';
+  import MemoizedMD from '@/markdown/MemoizedMD.svelte';
+  import ChevronRight from '@/icons/ChevronRight.svelte';
+  import Input from '@/lib/components/ui/input/input.svelte';
+
+  const { input, handleSubmit, messages } = useChat();
+</script>
+
+<div>
+  <div>
+    <form on:submit="{handleSubmit}">
```

Comment:
are we sure that wrapping dynamic value in double quote works with Svelte?

Reply:
I am not, @SferaDev @snide any idea?

Reply:
Wonder how the double quotes were inserted though? I saw that in the lint commit you did on the #201 PR...

Reply:
Hmm odd, it must have automatically formatted. I'll adjust now to single quotes. Any other similar quoting issues you see with the code?

Reply:
thank you! I think prettier re-inserted them so I'm evaluating this if it actually works!